components/eam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -0,0 +1,1667 @@
+    real(rtype), pointer :: cvreffliq(:,:)    ! convective cloud liquid effective radius (um)
+    real(rtype), pointer :: cvreffice(:,:)    ! convective cloud ice effective radius (um)
+    !! radiation 
+    real(rtype), pointer :: dei(:,:)          ! Ice effective diameter (um)
+    real(rtype), pointer :: mu(:,:)           ! Size distribution shape parameter for radiation
+    real(rtype), pointer :: lambdac(:,:)      ! Size distribution slope parameter for radiation
+    ! DONE PBUF
+    ! For recording inputs/outputs to p3_main
+    real(rtype) :: p3_main_inputs(pcols,pver+1,17) ! Record of inputs for p3_main
+    real(rtype) :: p3_main_outputs(pcols,pver+1,31) ! Record of outputs for p3_main
+
+    ! Derived Variables
+    real(rtype) :: icimrst(pcols,pver) ! stratus ice mixing ratio - on grid
+    real(rtype) :: icwmrst(pcols,pver) ! stratus water mixing ratio - on grid
+    real(rtype) :: rho(pcols,pver)
+    real(rtype) :: drout2(pcols,pver)
+    real(rtype) :: reff_rain(pcols,pver)
+    real(rtype) :: col_location(pcols,3),tmp_loc(pcols)  ! Array of column lon (index 1) and lat (index 2)
+    integer     :: tmpi_loc(pcols) ! Global column index temp array
+
+    ! Variables used for microphysics output
+    real(rtype) :: aqrain(pcols,pver)
+    real(rtype) :: anrain(pcols,pver)
+    real(rtype) :: nfice(pcols,pver)
+    real(rtype) :: efcout(pcols,pver)      
+    real(rtype) :: efiout(pcols,pver)      
+    real(rtype) :: ncout(pcols,pver)      
+    real(rtype) :: niout(pcols,pver)      
+    real(rtype) :: freqr(pcols,pver)      
+    real(rtype) :: freql(pcols,pver)      
+    real(rtype) :: freqi(pcols,pver)      
+    real(rtype) :: cdnumc(pcols)      
+    real(rtype) :: icinc(pcols,pver) 
+    real(rtype) :: icwnc(pcols,pver) 
+
+    ! variables for the CNT primary / heterogeneous freezing
+    real(rtype), pointer :: frzimm(:,:)
+    real(rtype), pointer :: frzcnt(:,:)
+    real(rtype), pointer :: frzdep(:,:)
+    real(rtype) :: frzimm_in(pcols,pver)
+    real(rtype) :: frzcnt_in(pcols,pver)
+    real(rtype) :: frzdep_in(pcols,pver)
+ 
+    integer :: it                      !timestep counter                       -
+    integer :: its, ite                !horizontal bounds (column start,finish)
+    integer :: kts                     !closest level to TOM                   -
+    integer :: kte                     !near surface level                     -
+
+    logical :: do_predict_nc           !prognostic droplet concentration or not?
+    logical :: do_subgrid_clouds       !use subgrid cloudiness in tendency calculations?
+    integer :: icol, ncol, k
+    integer :: psetcols, lchnk
+    integer :: itim_old
+
+    ! For rrtmg optics. specified distribution.
+    real(rtype), parameter :: dcon   = 25.e-6_rtype         ! Convective size distribution effective radius (um)
+    real(rtype), parameter :: mucon  = 5.3_rtype            ! Convective size distribution shape parameter
```

Comment:
The effective radius and effective diameter use different units? It says micron for both in the notes.

Reply:
The unit in the note here should be m, not um.

Reply:
It still looks suspicious to me. The two variables are used in the same equation but in different units: 
lambdac = (mucon + 1._rtype)/dcon.

Reply:
I did not check this line. Let me check and get back